api/node/__test__/eventloop.spec.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-1.1 OR LicenseRef-Slint-commercial
+
+// Test that the Slint event loop processes libuv's events.
+
+import test from 'ava'
+import * as http from 'http';
+import fetch from "node-fetch";
+
+import { run_event_loop, quit_event_loop, private_api } from '../index'
+
+
+test.serial('merged event loops with timer', async (t) => {
+
+    let invoked = false;
+
+    await run_event_loop(() => {
+        invoked = true;
+
+        setTimeout(() => {
+            quit_event_loop();
+        }, 2);
+    });
+    t.true(invoked)
+})
+
+
+test.serial('merged event loops with networking', async (t) => {
+    const port = 8000;
```

Comment:
It would be better not to hardcode the port here.
You can call [`server.listen`](https://nodejs.org/api/net.html#serverlistenport-host-backlog-callback) without a port:

> If port is omitted or is 0, the operating system will assign an arbitrary unused port, which can be retrieved by using server.address().port after the ['listening'](https://nodejs.org/api/net.html#event-listening) event has been emitted.

So the test works if there are other test running in parallel that would like to use that port number. (hopefully there isn't because no tests should hardcode ports :-))

Reply:
Good point. Done. There's one caveat: There might be prompts resulting from this. When running this on macOS I get one. But as long as it works in the CI that's fine.